src/routes/posts.js

```diff
@@ -20,8 +20,16 @@ const router = express.Router()
 // GET ALL ARTICLES
 
 const fetchPosts = catchAsync( async(req, res) => {
+	let feed
+	const {isFlagged} = req.query
 
-	const feed = await postService.fetchPosts()
+	
+	if (isFlagged === 'true' && req.user.role === 'admin'){
```

Comment:
also the logic should return a response of unauthorized if user is not an admin and the `isFlagged` query param is passed, like the issue says

Reply:
kindly look into applying the `isAdmin` middleware we have based on the existence of the `isFlagged` query param